### DISCUSSION TITLE
Confused about how to have ZZ cause close_window only when no views are open

### DISCUSSION THREAD

**Discussion by nikhilm** [NeoVintageous/NeoVintageous#953] [Q&A] [answered]
Thanks for an awesome plugin!

So, by default, `ZZ` is mapped to the `ex_exit` command, and that function first closes the current view, then, if there are no more open views in the window, it sends the window the `exit` command.

I don't like this because `exit` quits Sublime Text, when I want it to only close the current window. The relevant ST command is `close_window`.

However, if I do a regular rc file based remap of `ZZ` to `:CloseWindow<CR>`, then it no longer respects the "first close the current view" behavior.
Is there a way to script something like that?

**Comment by gerardroche** [comment, marked as answer]
We have a setting to change the behaviour of commands like ZZ:

```
    // Controls the behaviour of commands like ZZ, :quit, :wq, etc. When set to
    // `true`, quitting the last view will also exit Sublime Text. When set to
    // `false`, Sublime Text remains open after closing the last view.
    "vintageous_exit_when_quitting_last_window": true,
```

However, there are a few issues that will be fixed in the next version. 

1. There was a typo in the setting "vintageous_exit_when_quitting_last_window". 

    The "quitting" part only had one t. In the next version both setting will work, but two t's will be the correct setting: "vintageous_exit_when_quitting_last_window".
    
     In the current version you will need to use the misspelling: `vintageous_exit_when_quiting_last_window` but in the next version use the correctly spelled one. The new version will be released in the next few days. Maybe even later today.

2. ZZ was not respecting the setting. In the next version it will be the same as :wq

    In the meantime you can do a mapping:

    ```
    nnoremap ZZ :wq<CR>
    ```


Commands like :wq and :quit close the current window and if it's the last window then Vim exits.

Vim doesn't have a setting like the one above because in Vim you typically don't need to stay open if it's the last window. For instance, the side bar in Vim is considered a window, so if you have that open, then ZZ will close the file but Vim stays open because the side bar is now the last window. I've added an issue to emulate this behaviour too: https://github.com/NeoVintageous/NeoVintageous/issues/954

**Reply by nikhilm** [reply]
Thanks!

**Comment by gerardroche** [comment]
Added "unless_sidebar_visible" option:

```

    // Configures the behavior of commands like ZZ, :quit, :wq, etc.:
    // - Setting it to `true` makes quitting the last view exit Sublime Text.
    // - Setting it to `false` keeps Sublime Text open after closing the last view.
    // - Setting it to "unless_sidebar_visible" exits only if the sidebar is hidden.
    "vintageous_exit_when_quitting_last_window": true,
```
